### PR TITLE
bradl3yC - 6213 - Google analytics FE work address validation

### PR DESCRIPTION
**Pull request by bradl3yC**
## Description
Add dataLayer pushes to each of the interactions with address validation

### User updates address - address validation modal is shown with suggestions
Screen Shot 2020-03-06 at 10 14 51 AM

### Saved address via validation modal using OVERRIDE
Screen Shot 2020-03-06 at 10 30 54 AM

### Saved address via validation modal using SUGGESTED ADDRESS
Screen Shot 2020-03-06 at 10 31 55 AM

### Edit address link clicked
Screen Shot 2020-03-06 at 10 32 58 AM

### Address validation error
Screen Shot 2020-03-06 at 10 33 57 AM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx [hunk 1]
```diff
@@ -224,11 +233,7 @@ class AddressValidationModal extends React.Component {
           status="warning"
           headline={addressValidationMessage.headline}
         >
-          <addressValidationMessage.ModalText
-            editFunction={() =>
-              this.props.openModal(addressValidationType, addressFromUser)
-            }
-          />
+          <addressValidationMessage.ModalText editFunction={this.onEditClick} />
```
Comment (kelsonic) [inline src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: Easier to read, nicely done!

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉 Just to confirm, looks like with this PR we'll have these events being logged now:

```js
{
    event: 'profile-transaction',
    'profile-section': analyticsSectionName,
    'profile-addressSuggestionUsed': suggestedAddressSelected ? 'yes' : 'no',
}

{
      event: 'profile-navigation',
      'profile-action': 'edit-link',
      'profile-section': analyticsSectionName,
    }

{
      event: 'profile-edit-failure',
      'profile-action': 'address-suggestion-failure',
      'profile-section': analyticsSectionName,
    }

{
      event: 'profile-navigation',
      'profile-action': 'update-button',
      'profile-section': analyticsSectionName,
      'profile-addressValidationAlertShown': showModal ? 'yes' : 'no',
      'profile-addressSuggestionProvided': confirmedSuggestions.length
        ? 'yes'
        : 'no',
    }
```

Are there any others we want to include? 🙂

File: src/platform/user/profile/vet360/actions/transactions.js [hunk 2]
```diff
@@ -248,6 +248,17 @@ export const validateAddress = (
     // and only one confirmed address came back from the API
     const showModal = showAddressValidationModal(suggestedAddresses);
 
+    // push data to dataLayer for analytics
+    window.dataLayer.push({
```
Comment (erikphansen) [inline src/platform/user/profile/vet360/actions/transactions.js]: We should probably use the `recordEvent` helper for a little bit of safety. Would be a super quick change to make.

https://github.com/department-of-veterans-affairs/vets-website/blob/c72006cf2836bfbdf5895d28d9cb011d7a35dd03/src/platform/monitoring/record-event.js#L8

**Review by erikphansen — APPROVED** [review]

**Comment by bradl3yC** [comment]
@department-of-veterans-affairs/frontend-review-group ready for review 👍

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx [hunk 3]
```diff
@@ -35,16 +36,25 @@ class AddressValidationModal extends React.Component {
       addressValidationType,
       selectedAddress,
       selectedAddressId,
+      analyticsSectionName,
```
Comment (billfienberg) [inline src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: `analyticsSectionName` is a new prop, correct? If so, it's not actually being used anywhere yet, correct?

Comment (bradl3yC) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: It is on line 53, 63 and 71 - its also used on line 85

Comment (billfienberg) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: I see that `analyticsSectionName` is being used in the `AddressValidationModal` component. But this PR does not include updating the other files where the `AddressValidationModal` is rendered to include the new `analyticsSectionName` prop, correct?

Comment (bradl3yC) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: @billfienberg its being added to the component via mapStateToProps - I am not sure why the other components that render the modal would need to know about this prop.

Comment (kelsonic) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: `src/platform/user/profile/vet360/containers/AddressValidationModal.jsx:278+292` 🙂

Comment (billfienberg) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: There it is. Thank you. Sorry I didn't look more thoroughly to find it.

**Review by billfienberg — APPROVED** [review]
Makes sense to me

**Pull request merged**